tensorflow/contrib/seq2seq/python/ops/attention_wrapper.py

```diff
@@ -187,9 +187,12 @@ def __init__(self,
           "memory_layer is not a Layer: %s" % type(memory_layer).__name__)
     self._query_layer = query_layer
     self._memory_layer = memory_layer
+    self.dtype = memory_layer.dtype
     if not callable(probability_fn):
       raise TypeError("probability_fn must be callable, saw type: %s" %
                       type(probability_fn).__name__)
+    if score_mask_value is None:
+      score_mask_value = self._memory_layer.dtype.as_numpy_dtype(-np.inf)
```

Comment:
looks like self._memory_layer.dtype is a string, when it should be a tf dtype object.  looks like you need to use `dtypes.as_dtype(self._memory_layer.dtype).as_numpy_dtype(-np.inf)`
where `dtypes` is a module in python.framework.

sorry.